src/fragments/lib-v1/ssr/nextjs/getting-started.mdx

```diff
@@ -0,0 +1,335 @@
+    ssr: true, // required when using Amplify with Next.js
+});
+
+export default function RootLayoutThatConfiguresAmplifyOnTheClient({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  return children;
+}
+```
+
+<Callout>
+
+**TIP:** To avoid repetitive calls to `Amplify.configure`, you can call it once
+in a top-level client-side rendered layout component.
+
+</Callout>
+
+## Authentication
+
+You can use the Amplify Auth category APIs to sign up and sign in your end users
+on the client side. With setting `ssr: true` when calling `Amplify.configure`,
+the Amplify library uses cookies to store tokens, which will be sent along with
+HTTP requests to your Next.js app server.
+
+### Manage Auth Session with The `middleware`
+
+You can use the `fetchAuthSession` API to check the auth sessions that are attached
+to the incoming requests in the middleware of your Next.js app to protect your
+routes. For example:
+
+```typescript
+import { runWithAmplifyServerContext } from '@aws-amplify/adapter-nextjs';
+import { fetchAuthSession } from 'aws-amplify/auth/server';
+import { NextRequest, NextResponse } from 'next/server';
+
+export async function middleware(request: NextRequest) {
+  const response = NextResponse.next();
+
+  const authenticated = await runWithAmplifyServerContext({
+    nextServerContext: { request, response },
+    operation: async (contextSpec) => {
+      try {
+        const session = await fetchAuthSession(contextSpec, {});
+        return session.tokens !== undefined;
+      } catch (error) {
+        console.log(error);
+        return false;
+      }
+    },
+  });
+
+  if (authenticated) {
+    return response;
+  }
+
+  return NextResponse.redirect(new URL('/sign-in', request.url));
+}
+
+export const config = {
+  matcher: [
+    /*
+     * Match all request paths except for the ones starting with:
+     * - api (API routes)
+     * - _next/static (static files)
+     * - _next/image (image optimization files)
+     * - favicon.ico (favicon file)
+     */
+    '/((?!api|_next/static|_next/image|favicon.ico|sign-in).*)',
+  ],
+};
+```
+
+In this example, if the incoming request is not associated with a valid user
+session the request will be redirected to the `/sign-in` route.
+
+<Callout>
+
+**NOTE:** When calling `fetchAuthSession` with a `response` context, it will send the
+refreshed tokens (if any) back to the client via the `Set-Cookie` header in the response.
+
+</Callout>
+
+## Using Amplify Category APIs as a Data Source on The Server Side
+
+In general, to use Amplify APIs on the server in your Next.js app, you will
+need to:
+
+1. Import the API from the `/server` sub path.
+2. Using the `runWithAmplifyServerContext` helper function exported from
+`@aws-amplify/adapter-nextjs` to call the Amplify API in an isolated server context.
+
+<Callout>
+
+**NOTE:** A subset of Amplify APIs can now be called on the server side of a Next.js
+app. These APIs are exported from the `/server` sub paths. See
```

Comment:
I think we should make clear that they *must* use those APIs from `/server` sub path, something like may be on line 147:

```
You **must** use these APIs instead of normal APIs not from the `/server` sub path for APIs run on server-side.
```